TYPO3-Documentation/t3SphinxThemeRtd
# [FEATURE] Add new rendering theme

### How to Test:

1. Install 
    ```
    yarn install
    yarn build
    ```
    This will render the "How to Document Guide" as reference.
    If you want to test a different documentation, simply provide the path.
    ```
    # full build
    yarn build --source=..\t3SphinxThemeRtdDemoDocs

    # just render
    yarn render --source=..\t3SphinxThemeRtdDemoDocs
    ```

2. Start HTTP-Server
    ```
    yarn serve
    ```
3. Validate
    Open Browser on the Served Address and check if everything is working

### Testing

Each change to the theme will trigger a GitHub action build, that will provide an artifact of a rendered documentation of https://github.com/TYPO3-Documentation/TYPO3CMS-Guide-HowToDocument. See: https://github.com/benjaminkott/typo3-sphinx-theme/actions

1. 
![image](https://user-images.githubusercontent.com/3243119/71807915-b933c380-306c-11ea-9af3-814659241d88.png)
2. 
![image](https://user-images.githubusercontent.com/3243119/71807960-d8325580-306c-11ea-9ee4-58a1d268d972.png)



### Fixes: 

- **#122 Contrast for some syntax highlighting colors - readability could be improved** 
  A new color theme is in place.

- **#110 Color scheme (orange) of docs.typo3.org inconsistent with TYPO3 style guide**
  Streamlined for now, will be discussed with design team.

- **#109 Using a sidebar at end of page breaks rendering**
  Sidebar does not break the rendering anymore.

- **#104 Improve readability by adding more space around images**
  Image display is now streamlined with margins.

- **#103 Make toolchain and documentation for toolchain "contributor-ready"**
  **#85 Building t3SphinxThemeRtd fails with 404**
  Theme development is now reduced to the following steps, it will always take the current installed documentation docker container or build pull a new one.
  ```
  yarn install
  yarn build
  yarn serve
  ```
- **#92 Changed urls for links in contentinfo section (bottom)**
  Links are now updated.

- **#94 Add homepage link to breadcrumb and change anchor text for start of manual**
  Breadcrumb root node was renamed to `Start`

- **#89 Dimmed buttons should be better visible - a bit**
  Pager on top was removed.

- **#81 Restyle and restructure left navigation panel**
  Navigation panel was rebuild from the ground up.

- **#141 Next / prev page causes new page to "jump" while rendered**
  Introduced a hidden navigation on top that is not displayed and not in the tabindex,
  this navigation is only used to provide targets for previous and next access key navigation.
  This prevents jumping to the bottom navigation when using the shortcuts.

- **#133 Position "Related links" correctly**
  Related Links bar was reworked and does not exist anymore in its current form.
  The version selector and contents have been moved to different locations.

- **#118 Release version is not displayed on mobile**
  Version/Release is not hidden anymore on mobile and always visible.

- **#64 Color change for :hover not working for visited links**
  Links are now changing their hover color correctly.

- **#138 Long inline code will be displayed without line breaking and ends up outside of viewport**
  Long inline code lines are now breaking into the next row when they are too long.
